### ISSUE TITLE
Dell XPS 9760 "Only 1 GPUs found, there need ot be at least 2"

### ISSUE THREAD

**Issue by e-compton** [hertg/egpu-switcher#3]
Using a Gigabyte AUROS NVIDIA 1080 and the Dell XPS 9760 I'm getting the following error:
```
[warn] Only 1 GPUs found, there need to be at least 2. Make sure to connect your EGPU for the setup.
```
Btw, this is on Ubuntu 18.04.

I imagine it should detect the connected 1080 and the intel intregrated graphics. The eGPU was definitely connected and recognised. At the time of running the eGPU was outputting to an external display (activated using prime-select nvidia).

Does nvidia-prime have to be somehow uninstalled/ deactivated for this to work?

**Comment by hertg** [comment]
@e-compton 
Thanks for the report.
It should indeed recognize your EGPU if it is connected and the Thunderbolt 3 connection is authorized.

Seems like it doesn't output your EGPU when executing the following command?
```bash
lspci | grep -i -E "vga|3d"
```

If not, could you please execute `lspci` without grep and post if your EGPU shows up in there?

**Comment by e-compton** [comment]
Output of `lspci | grep -i -E "vga|3d"`:
```
00:02.0 VGA compatible controller: Intel Corporation Device 3ea0
3c:00.0 VGA compatible controller: NVIDIA Corporation GP104 [GeForce GTX 1080] (rev a1)
3d:00.0 USB controller: Intel Corporation DSL6540 USB 3.1 Controller [Alpine Ridge]
```

**Comment by e-compton** [comment]
`lspci`:
```
00:00.0 Host bridge: Intel Corporation Device 3e34 (rev 0b)
00:02.0 VGA compatible controller: Intel Corporation Device 3ea0
00:04.0 Signal processing controller: Intel Corporation Xeon E3-1200 v5/E3-1500 v5/6th Gen Core Processor Thermal Subsystem (rev 0b)
00:08.0 System peripheral: Intel Corporation Xeon E3-1200 v5/v6 / E3-1500 v5 / 6th/7th Gen Core Processor Gaussian Mixture Model
00:12.0 Signal processing controller: Intel Corporation Device 9df9 (rev 30)
00:14.0 USB controller: Intel Corporation Device 9ded (rev 30)
00:14.2 RAM memory: Intel Corporation Device 9def (rev 30)
00:15.0 Serial bus controller [0c80]: Intel Corporation Device 9de8 (rev 30)
00:15.1 Serial bus controller [0c80]: Intel Corporation Device 9de9 (rev 30)
00:16.0 Communication controller: Intel Corporation Device 9de0 (rev 30)
00:1c.0 PCI bridge: Intel Corporation Device 9dbc (rev f0)
00:1c.6 PCI bridge: Intel Corporation Device 9dbe (rev f0)
00:1d.0 PCI bridge: Intel Corporation Device 9db0 (rev f0)
00:1d.4 PCI bridge: Intel Corporation Device 9db4 (rev f0)
00:1f.0 ISA bridge: Intel Corporation Device 9d84 (rev 30)
00:1f.3 Audio device: Intel Corporation Device 9dc8 (rev 30)
00:1f.4 SMBus: Intel Corporation Device 9da3 (rev 30)
00:1f.5 Serial bus controller [0c80]: Intel Corporation Device 9da4 (rev 30)
01:00.0 Unassigned class [ff00]: Realtek Semiconductor Co., Ltd. RTS525A PCI Express Card Reader (rev 01)
02:00.0 Network controller: Qualcomm Atheros QCA6174 802.11ac Wireless Network Adapter (rev 32)
03:00.0 PCI bridge: Intel Corporation JHL6540 Thunderbolt 3 Bridge (C step) [Alpine Ridge 4C 2016] (rev 02)
04:00.0 PCI bridge: Intel Corporation JHL6540 Thunderbolt 3 Bridge (C step) [Alpine Ridge 4C 2016] (rev 02)
04:01.0 PCI bridge: Intel Corporation JHL6540 Thunderbolt 3 Bridge (C step) [Alpine Ridge 4C 2016] (rev 02)
04:02.0 PCI bridge: Intel Corporation JHL6540 Thunderbolt 3 Bridge (C step) [Alpine Ridge 4C 2016] (rev 02)
04:04.0 PCI bridge: Intel Corporation JHL6540 Thunderbolt 3 Bridge (C step) [Alpine Ridge 4C 2016] (rev 02)
05:00.0 System peripheral: Intel Corporation JHL6540 Thunderbolt 3 NHI (C step) [Alpine Ridge 4C 2016] (rev 02)
39:00.0 USB controller: Intel Corporation JHL6540 Thunderbolt 3 USB Controller (C step) [Alpine Ridge 4C 2016] (rev 02)
3a:00.0 PCI bridge: Intel Corporation DSL6540 Thunderbolt 3 Bridge [Alpine Ridge 4C 2015]
3b:01.0 PCI bridge: Intel Corporation DSL6540 Thunderbolt 3 Bridge [Alpine Ridge 4C 2015]
3b:04.0 PCI bridge: Intel Corporation DSL6540 Thunderbolt 3 Bridge [Alpine Ridge 4C 2015]
3c:00.0 VGA compatible controller: NVIDIA Corporation GP104 [GeForce GTX 1080] (rev a1)
3c:00.1 Audio device: NVIDIA Corporation GP104 High Definition Audio Controller (rev a1)
3d:00.0 USB controller: Intel Corporation DSL6540 USB 3.1 Controller [Alpine Ridge]
6e:00.0 Non-Volatile memory controller: SK hynix Device 1527
```

**Comment by hertg** [comment]
@e-compton
Sorry for the late response.
Your output looks good, it shouldn't output your USB controller, but thats another bug in the script (I created the issue #4).

Unfortunately i can't explain whats happening on your setup.
The script executes the `lspci | grep -i -E "vga|3d"` command, just as you manually did. So it should find the GTX 1080.

Could you please re-run the `egpu-switcher setup` command, and then post the results of the following command. 

```bash
cat /tmp/egpu-switcher/results
```

> Thats where the results for the found GPUs are stored. The message `[warn] Only 1 GPUs found, there need to be at least 2. Make sure to connect your EGPU for the setup.
` appears if theres only **one line** of text in this file.

**Comment by hertg** [comment]
closing due to inactivity

**Comment by tristan-k** [comment]
Please re-open the issue. I'm running a `Lenovo YOGA 730-13IKB` with a GTX 1060 6GB in a ZOTAC AMP BOX MINI and I'm stuck with the same issue. My IntelUHD will not be picked up.

```
➜  ~ boltctl list

 ● ZOTAC AMP BOX MINI
   ├─ type:          peripheral
   ├─ name:          AMP BOX MINI
   ├─ vendor:        ZOTAC
   ├─ uuid:          XXXXXXXXXXXXXXXXXXX
   ├─ status:        authorized
   │  ├─ domain:     domain0
   │  └─ authflags:  none
   ├─ authorized:    So 23 Jun 2019 11:31:33 UTC
   ├─ connected:     So 23 Jun 2019 11:31:33 UTC
   └─ stored:        Sa 22 Jun 2019 12:51:29 UTC
      ├─ policy:     auto
      └─ key:        no
```

```
➜  ~ nvidia-smi 
Sun Jun 23 13:34:25 2019       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 430.26       Driver Version: 430.26       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 106...  Off  | 00000000:06:00.0 Off |                  N/A |
| 44%   30C    P0    25W / 120W |      0MiB /  6078MiB |      3%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

```
➜  ~ lspci | grep -i -E "vga|3d"
00:02.0 VGA compatible controller: Intel Corporation UHD Graphics 620 (rev 07)
06:00.0 VGA compatible controller: NVIDIA Corporation GP106 [GeForce GTX 1060 6GB] (rev a1)         

➜  ~ sudo egpu-switcher setup      
[info] No argument passed to read_gpus(). Using nvidia-xconfig by default.
[warn] Only 1 GPUs found, there need to be at least 2. Make sure to connect your EGPU for the setup.

➜  ~ cat /tmp/egpu-switcher/results
GeForce GTX 1060 6GB (PCI:6:0:0)
```

**Comment by tristan-k** [comment]
I did some digging and it seems as the version in your ppa is outdated and doesn't incorporate the latest changes. Furthermore after manually download the bionic version and installing it with `gdebi` I discovered an issue with the systemd service. 

```
➜  ~ sudo egpu-switcher setup            
[info] Using default argument for read_gpus().

Found 2 possible GPUs...

  1: Intel Corporation UHD Graphics 620 (rev 07) (PCI:0:2:0)
  2: NVIDIA Corporation GP106 [GeForce GTX 1060 6GB] (rev a1) (PCI:6:0:0)

Choose your preferred INTERNAL GPU [1-2]: 1
Choose your preferred EXTERNAL GPU [1-2]: 2

[info] Using default argument for is_egpu_connected().
[info] Automatically detecting if egpu is connected...  true
[info] Create symlink /etc/X11/xorg.conf -> /etc/X11/xorg.conf.egpu
Created symlink /etc/systemd/system/multi-user.target.wants/egpu.service → /etc/systemd/system/egpu.service.
[success] Done... Setup finished

➜  ~ sudo systemctl status egpu.service
● egpu.service - EGPU Service
   Loaded: error (Reason: Exec format error)
   Active: inactive (dead)

Jun 23 13:52:01 tk-ly730-13IKB-ubuntu systemd[1]: /etc/systemd/system/egpu.service:5: Executable path is not absolute: egpu-switcher switch auto
```

**Comment by hertg** [comment]
@tristan-k 
Thanks for the input, i'll reopen the issue.
Yes, my latest releases were only published for disco, because its the only version that i'm testing.

So, did the latest (disco) version fix the issue with not detecting integrated graphics?
I have made some minor changes in the lspci detection but couldn't test them yet , since my egpu enclosure has some issues.

**Comment by hertg** [comment]
@tristan-k 
@e-compton 
I have published a new release **0.10.1**, for `disco` as well as `bionic`. Please update your installation with:
```bash
sudo apt update
sudo apt upgrade egpu-switcher
```

**You'll have to rerun the setup after that**.
```bash
sudo egpu-switcher setup
```

Please report back if your problem still persists.
